### PR TITLE
Add sticky messages

### PR DESCRIPTION
**Pull request by DEVTomatoCake**
Adds a sticky messages module, closes https://features.sc-network.net/board/custombot/532.
There were some random issues in testing that only occurred once, so I just hope they're fixed, hehe.

The package-lock.json seems to be updated automatically due to running `npm update` - should I revert the changes in the file? Shouldn't cause any issues though.

### CODE REVIEW

File: modules/sticky-messages/events/messageCreate.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * @param {Discord.Message} msg
+ * @param {Object} configMsg The configured message
+ */
+async function sendMessage(client, msg, configMsg) {
+    channelData[msg.channel.id] = {
+        msg: null,
+        timeout: null,
+        time: client.uptime
+    };
+    const sentMessage = await msg.channel.send(await embedTypeV2(configMsg || 'Message not found',
```
Comment (SCDerox) [inline modules/sticky-messages/events/messageCreate.js:31]: configMsg will never be null (or undefined), as this checked in line 54

File: modules/sticky-messages/events/messageCreate.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        {},
+        {},
+        []
```
Comment (SCDerox) [inline modules/sticky-messages/events/messageCreate.js:34]:
```suggestion
```
These are the default values, I don't see a technical requirement or restriction for the need to re-set them (and potentially overwrite future default value changes). embedTypeV2 has the same API as embedType (and simply calls it with the same values)

**Review by SCDerox — CHANGES_REQUESTED** [review]
Thanks so much for your contribution - I've found one small issue, please review my comment.

File: modules/sticky-messages/events/messageCreate.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    ));
+    channelData[msg.channel.id] = {
+        msg: sentMessage.id,
+        timeout: null,
+        time: client.uptime // Much smaller number compared to Date.now()
```
[the remark this answers is not in the document]
Comment (SCDerox) [reply modules/sticky-messages/events/messageCreate.js]: This was not a change request, you could have left it at client.uptime, I just wanted to explain why the number size does not matter

[the remark this answers is not in the document]
Comment (DEVTomatoCake) [reply modules/sticky-messages/events/messageCreate.js]: Yes, however - as you correctly stated - it doesn't make a difference, and as using a timestamp for dates is more common, it makes more sense to use it here too. Also it removes the need of using the client object as function parameter.

**Review by SCDerox — APPROVED** [review]
Thanks so much for your contribution, it will be available on SCNX shortly :rocket:

**Comment by SCDerox** [comment]
Thanks for your contribution - you have received a payout as part of our [Open Source Developer Program](https://faq.scnx.app/open-source-developer-pool/). You can find details in your SCNX organisation payout history. To pay out the received amount, please follow instructions in your dashboard. If you have any questions, please contact ScootKit staff on [scnx.app/help](https://scnx.app/help).
Cheers ✌️

**Pull request merged**